targets/ChibiOS/RP_PICO2_RP2350/nanoBooter/main.c

```
@@ -0,0 +1,88 @@
+
+    // The kernel is initialized but not started yet, this means that
+    // main() is executing with absolute priority but interrupts are already enabled.
+    osKernelInitialize();
+
+    // check if there is a request to remain on nanoBooter
+    if (!IsToRemainInBooter())
+    {
+        // check for valid CLR image at address contiguous to nanoBooter
+        // NOTE: The standard CheckValidCLRImage() opcode check fails on RP2350 due to XIP flash read
+        // behavior. Use a simple vector table validation instead.
+        volatile uint32_t *clrVector = (volatile uint32_t *)(uint32_t)&__nanoImage_end__;
```

Clang (14.0.6)

[warning] variable 'clrVector' is not initialized

(cppcoreguidelines-init-variables)